### ISSUE TITLE
.inc file included for lowest version only

### ISSUE THREAD

**Issue by offa** [priv-kweihmann/oelint-adv#246]
Linting multiple recipes which include a common .inc file reports missing variables for all files other than the lowest version. Linting each file by it's own doesn't report the issues.

#### Example

- _cppcheck_2.5.bb_
- _cppcheck_2.6.bb_
- _cppcheck.inc_ (common fields like summary, author, homepage etc.; included by both .bb files through `require ${PN}.inc`)

Executing an analysis reports eg. missing variables for all entries made in the .inc file:

```sh
oelint-adv --quiet cppcheck/cppcheck_2.*.bb

[...]/cppcheck/cppcheck_2.6.bb:1:error:oelint.var.mandatoryvar.SUMMARY:Variable 'SUMMARY' should be set
[...]/cppcheck/cppcheck_2.6.bb:1:error:oelint.var.mandatoryvar.HOMEPAGE:Variable 'HOMEPAGE' should be set
[...]/cppcheck/cppcheck_2.6.bb:1:info:oelint.var.suggestedvar.AUTHOR:Variable 'AUTHOR' should be set
[...]/cppcheck/cppcheck_2.6.bb:1:info:oelint.var.suggestedvar.CVE_PRODUCT:Variable 'CVE_PRODUCT' should be set
[...]/cppcheck/cppcheck_2.6.bb:1:error:oelint.var.mandatoryvar.DESCRIPTION:Variable 'DESCRIPTION' should be set
[...]/cppcheck/cppcheck_2.6.bb:1:error:oelint.var.mandatoryvar.LICENSE:Variable 'LICENSE' should be set
[...]/cppcheck/cppcheck_2.6.bb:1:info:oelint.var.suggestedvar.SECTION:Variable 'SECTION' should be set
[...]/cppcheck/cppcheck_2.6.bb:1:info:oelint.var.suggestedvar.BUGTRACKER:Variable 'BUGTRACKER' should be set
```

Everything is ok if each file is checked on it's own.

- If a 2.4 (lower) file is added: Warnings for 2.5 and 2.6
- If a 2.7 (higher) file is added: Warnings for 2.6 and 2.5

#### Setup

**- oelint-adv 3.2.0
- Python 3.8 / 3.10**

**Comment by priv-kweihmann** [comment]
Okay I'll have a look - either somethings going wrong with grouping the file or in the final deduplicating of the found issues
